run_page/keep_to_strava.py

```diff
@@ -0,0 +1,152 @@
+    keep2strava_bk_path = os.path.join(OUTPUT_DIR, "keep2strava.json")
+    if not os.path.exists(keep2strava_bk_path):
+        file = open(keep2strava_bk_path, "w")
+        file.close()
+        content = []
+    else:
+        with open(keep2strava_bk_path) as f:
+            try:
+                content = json.loads(f.read())
+            except:
+                content = []
+    old_tracks_ids = [str(a["run_id"]) for a in content]
+    new_tracks = get_all_keep_tracks(email, password, old_tracks_ids, with_download_gpx)
+
+    return new_tracks
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser()
+    parser.add_argument("phone_number", help="keep login phone number")
+    parser.add_argument("password", help="keep login password")
+    parser.add_argument("client_id", help="strava client id")
+    parser.add_argument("client_secret", help="strava client secret")
+    parser.add_argument("strava_refresh_token", help="strava refresh token")
+
+    options = parser.parse_args()
+    new_tracks = run_keep_sync(options.phone_number, options.password, True)
+
+    # to strava.
+    print("Need to load all gpx files maybe take some time")
+    last_time = 0
+    client = make_strava_client(
+        options.client_id, options.client_secret, options.strava_refresh_token
+    )
+
+    index = 1
+    print(f"Up to {len(new_tracks)} files are waiting to be uploaded")
+    uploaded_file_paths = []
+    for track in new_tracks:
+        if track.gpx_file_path is not None:
+            try:
+                upload_file_to_strava(client, track.gpx_file_path, "gpx", False)
+                uploaded_file_paths.append(track)
+            except RateLimitTimeout as e:
+                timeout = e.timeout
+                print(f"Strava API Rate Limit Timeout. Retry in {timeout} seconds\n")
+                time.sleep(timeout)
+                # try previous again
+                upload_file_to_strava(client, track.gpx_file_path, "gpx", False)
+                uploaded_file_paths.append(track)
+            except ActivityUploadFailed as e:
+                print(f"Upload faild error {str(e)}")
+            # spider rule
+            time.sleep(1)
+        else:
+            # for no gps data, like indoorRunning.
+            uploaded_file_paths.append(track)
+    time.sleep(10)
+
+    keep2strava_bk_path = os.path.join(OUTPUT_DIR, "keep2strava.json")
+    with open(keep2strava_bk_path, "r") as f:
+        try:
+            content = json.loads(f.read())
+        except:
+            content = []
+    content.extend(
+        [
+            dict(
+                run_id=track.id,
+                name=track.name,
+                type=track.type,
+                gpx_file_path=track.gpx_file_path,
+            )
+            for track in uploaded_file_paths
+        ]
+    )
+    with open(keep2strava_bk_path, "w") as f:
+        json.dump(content, f, indent=0)
+
+    # del gpx
+    for track in new_tracks:
```

Comment:
Since GPX files that fail to upload will still be uploaded next time, it might be better to delete them all to keep GPXOUT  clean.

Deleting files while uploading, I'm not sure if I'm going to have a problem similar to ”# Fix the issue that the github action runs too fast, resulting in unsuccessful file generation“ in nike_to_strava.py. If I need to add sleep to avoid it, then I might as well delete them last to keep the execution efficient.

Reply:
Adding comments would be helpful to clarify the code